test/BackgroundTaskTests/BackgroundTaskBuilderTests.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation and Contributors. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include <winrt/Windows.Foundation.h>
+#include <TerminalVelocityFeatures-BackgroundTask.h>
+#include <winrt\windows.storage.h>
+#include <winrt\windows.applicationmodel.background.h>
+#include <winrt\Microsoft.Windows.ApplicationModel.Background.h>
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+using namespace winrt::Windows::ApplicationModel::Background;
+using namespace winrt::Windows::Storage;
+
+namespace TB = ::Test::Bootstrap;
+namespace TP = ::Test::Packages;
+
+namespace BackgroundTaskTests
+{
+	class BackgroundTaskBuilderTests
+	{
+    public:
+        BEGIN_TEST_CLASS(BackgroundTaskBuilderTests)
+            TEST_CLASS_PROPERTY(L"ThreadingModel", L"MTA")
+            TEST_CLASS_PROPERTY(L"RunFixtureAs:Class", L"RestrictedUser")
+            TEST_CLASS_PROPERTY(L"RunAs", L"UAP")
+            TEST_CLASS_PROPERTY(L"UAP:AppxManifest", L"BackgroundTaskBuilder-AppxManifest.xml")
+            END_TEST_CLASS()
+
+        TEST_CLASS_SETUP(ClassInit)
+        {
+            ::Test::Bootstrap::SetupPackages();
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassUninit)
+        {
+            ::Test::Bootstrap::CleanupPackages();
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_WindowsAppRuntimeFramework());
+            return true;
+        }
+
+        TEST_METHOD_CLEANUP(MethodUninit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_WindowsAppRuntimeFramework());
+            return true;
+        }
+
+		TEST_METHOD(TestBackgroundTaskRegistration)
+		{
+            if (!::Microsoft::Windows::ApplicationModel::Background::Feature_BackgroundTask::IsEnabled())
```

Comment:
We can add more logs here for better diagnosability

Reply:
Please clarify on this more, We already have logs for `IsEnabled` check and for other assertion checks there will be logs created as well

Reply:
Agree. But you can consider adding comments using Log::Comment mentioning what is happening in the test. This would give more clarity on test status. But this is optional.

Reply:
Test names are quite self explanatory, Will add later if required